modules/social_features/social_group/social_group.module

```diff
@@ -2322,29 +2328,33 @@ function social_group_can_view_groups_of_type(string $type, AccountInterface $ac
 
   // Anonymous users have a special role.
   if ($account->isAnonymous()) {
-    /** @var \Drupal\group\Entity\GroupRoleInterface $role */
-    return $group_type->getAnonymousRole()->hasPermission('view group');
+    /** @var \Drupal\group\Entity\GroupRoleInterface $anonymous_role */
+    $anonymous_role = \Drupal::entityTypeManager()->getStorage('group_role')
+      ->load($group_type->id() . '-anonymous');
+    return $anonymous_role->hasPermission('view group');
   }
 
-  // The service that keeps platform roles in sync with group roles for
-  // outsiders.
-  /** @var \Drupal\group\GroupRoleSynchronizerInterface $group_role_synchroniser */
-  $group_role_synchroniser = \Drupal::service('group_role.synchronizer');
   /** @var \Drupal\group\Entity\GroupRoleInterface[] $group_roles */
   $group_roles = $group_type->getRoles();
 
   $user_roles = $account->getRoles(TRUE);
 
   // Authenticated is a locked Role,
   // So first we check outsiders also easier on the performance :)
-  $outsider = $group_type->getOutsiderRole();
+  /** @var \Drupal\group\Entity\GroupRoleInterface $outsider */
+  $outsider = \Drupal::entityTypeManager()
+    ->getStorage('group_role')
+    ->load($group_type->id() . '-outsider');
+
   if (NULL !== $outsider && $outsider->hasPermission('view group')) {
     return TRUE;
   }
 
   // If members can see it, we also need to show group types because
   // we are not calculation memberships.
-  $member = $group_type->getMemberRole();
+  $id = $group_type->id() . '-member';
```

Comment:
Couldn't we do these things by scope? for example by `scope: insider` for example? that gives us an easier way to re-use things across other parts, instead of having to name these roles using this group_type - member format

Reply:
Replaced with scope check